kotlin/injivcrenderer/src/commonMain/kotlin/io/mosip/injivcrenderer/qrCode/QrCodeGenerator.kt

```
@@ -2,24 +2,27 @@ package io.mosip.injivcrenderer.qrCode
-            return convertQrDataIntoBase64(qrData)
-
+            return generateQrCodeFromVcJson(vcJson)
         } catch (e: Exception){
             throw VcRendererExceptions.QRCodeGenerationFailureException(traceabilityId, e.message ?: UNKNOWN_ERROR,  className)
         }
     }
 
+    fun generateFromQrData(qrData: String): String {
+        try {
+            return generateQrCodeFromQrData(qrData)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)